wpcom-legacy-redirector.php

```diff
@@ -140,14 +145,14 @@ static function get_redirect_uri( $url ) {
 	}
 
 	static function get_redirect_post_id( $url ) {
-		global $wpdb;
-
 		$url_hash = self::get_url_hash( $url );
 
-		$redirect_post_id = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_type = %s AND post_name = %s LIMIT 1", self::POST_TYPE, $url_hash ) );
+		$redirect_post = get_page_by_path( $url_hash, OBJECT, array( self::POST_TYPE ) );
```

Comment:
We're intentionally doing a direct database call here; would prefer not to change this, especially since it adds the extra overhead that comes with `get_page_by_path`.